File: WPF/MVVM/Converters/TimeSpanFriendlyConverter.vb
Imports System.Globalization
Imports System.Windows.Data

Namespace Common.MVVM.Converters

    Public Class TimeSpanFriendlyConverter
        Implements IValueConverter

        Public Function Convert(ByVal Value As Object, ByVal TargetType As Type, ByVal Parameter As Object, ByVal Culture As CultureInfo) As Object Implements IValueConverter.Convert
            Return Utilities.Representation.GetFriendlyTimeSpan(CType(Value, TimeSpan), TimeSpan.FromMinutes(CType(Parameter, Double)))
        End Function

        Public Function ConvertBack(ByVal Value As Object, ByVal TargetType As Type, ByVal Parameter As Object, ByVal Culture As CultureInfo) As Object Implements IValueConverter.ConvertBack
            ' ToDo Implement the reverse conversion.
            Throw New NotSupportedException()
        End Function

    End Class

End Namespace

File: WPF/MVVM/Converters/TimeSpanFriendlyConverter.cs
using System;
using System.Globalization;
using System.Windows.Data;

namespace Ks.Common.MVVM.Converters
{
        public class TimeSpanFriendlyConverter : IValueConverter
        {
            public object Convert(object Value, Type TargetType, object Parameter, CultureInfo Culture)
            {
                return Utilities.Representation.GetFriendlyTimeSpan((TimeSpan)Value, TimeSpan.FromMinutes(System.Convert.ToDouble(Parameter)));
            }

            public object ConvertBack(object Value, Type TargetType, object Parameter, CultureInfo Culture)
            {
                // ToDo Implement the reverse conversion.
                throw new NotSupportedException();
            }
        }
    }
